### DISCUSSION TITLE
Why I cannot use 'from sqlalchemy import mapped_column'?

### DISCUSSION THREAD

**Discussion by WendaoLee** [sqlalchemy/sqlalchemy#8634] [Usage Questions] [answered]
In the [docs](https://docs.sqlalchemy.org/en/20/orm/declarative_styles.html#declarative-mapping-using-a-decorator-no-declarative-base),there has code `from sqlalchemy import mapped_column`.But it cannot be recognized by my IDE:
<img width="249" alt="image" src="https://user-images.githubusercontent.com/39958730/195753086-07087dd7-8e33-41ef-8401-40c1f4249f96.png">

I'm sure I have installed the 2.0 version:
<img width="221" alt="image" src="https://user-images.githubusercontent.com/39958730/195753305-2776da3f-a0ed-41ea-a73f-d8341f979e55.png">

This makes me confused.Is this IDE's error or document's error?😢

**Comment by CaselIT** [comment]
Hi,

It's an ide error. You should also uninstall any installed sqlalchemy stubs package

**Reply by CaselIT** [reply]
Sorry I had misread the import and read it as `from sqlalchemy.orm import mapped_column`

**Reply by WendaoLee** [reply]
> Sorry I had misread the import and read it as `from sqlalchemy.orm import mapped_column`

That's not a big case :) When you replied me I was not at home so I did't pay much time on it.Thank you for your reply!

**Comment by zzzeek** [comment, marked as answer]
that's a bug in our docs!  will fix ASAP

should be

```py
from sqlalchemy import ForeignKey, Integer, String, Text
from sqlalchemy.orm import mapped_column, registry, relationship
```

**Reply by WendaoLee** [reply]
Thank you for your explanation!🥰

**Reply by Ducky6944** [reply]
Any Idea what is happening here?
```PYTHON
from sqlalchemy import Column, Date, Integer, Float, String, ForeignKey, delete
from sqlalchemy.ext.asyncio import AsyncSession, create_async_engine
from sqlalchemy.orm import declarative_base, sessionmaker, relationship, mapped_column
from sqlalchemy.future import select
```
```LOGS
from sqlalchemy.orm import mapped_column
ImportError: cannot import name 'mapped_column' from 'sqlalchemy.orm' (/home/typkrft/.local/lib/python3.10/site-packages/sqlalchemy/orm/__init__.py) 
```
`sqlalchemy-1.4.46`

**Reply by CaselIT** [reply]
mapped column is a v2 thing, so it's not available in 1.4 series

**Reply by Ducky6944** [reply]
> mapped column is a v2 thing, so it's not available in 1.4 series

I just read that in a bulletin. Thanks for the reply thought I was a little confused when a lot of the documentation kind of alludes to 2.0 style in 1.4.

**Comment by zzzeek** [comment]
That's fixed in d78685b9ce3e88b14d48db8d3a.   the correct example should be:

```py
from datetime import datetime
from typing import Optional

from sqlalchemy import ForeignKey
from sqlalchemy import func
from sqlalchemy import Integer
from sqlalchemy import String
from sqlalchemy.orm import Mapped
from sqlalchemy.orm import mapped_column
from sqlalchemy.orm import registry
from sqlalchemy.orm import relationship

mapper_registry = registry()


@mapper_registry.mapped
class User:
    __tablename__ = "user"

    id = mapped_column(Integer, primary_key=True)
    name: Mapped[str]
    fullname: Mapped[Optional[str]]
    nickname: Mapped[Optional[str]] = mapped_column(String(64))
    create_date: Mapped[datetime] = mapped_column(insert_default=func.now())

    addresses: Mapped[list["Address"]] = relationship(back_populates="user")


@mapper_registry.mapped
class Address:
    __tablename__ = "address"

    id = mapped_column(Integer, primary_key=True)
    user_id = mapped_column(ForeignKey("user.id"))
    email_address: Mapped[str]

    user: Mapped["User"] = relationship(back_populates="addresses")

```
sorry for the confusion

**Comment by elysenko** [comment]
Will the mapped_column ever work for python 3.9?

https://stackoverflow.com/questions/78348018/sqlalchemy-mapped-column-wasnt-found

**Reply by zzzeek** [reply]
python 3.9 is supported and there are no known issues
